contracts/vaults/Base.sol

```diff
@@ -201,171 +126,157 @@ contract Base is Ownable, ReentrancyGuard {
+    event SetMaxBounty(uint256 _maxBounty);
+    event SetRewardController(IRewardController indexed _newRewardController);
+    event UpdateVaultInfo(
         bool indexed _registered,
         bool _depositPause,
         string _descriptionHash
     );
-    event SwapAndBurn(
-        uint256 indexed _pid,
-        uint256 indexed _amountSwapped,
-        uint256 indexed _amountBurned
-    );
-    event SwapAndSend(
-        uint256 indexed _pid,
-        address indexed _beneficiary,
-        uint256 indexed _amountSwapped,
-        uint256 _amountReceived,
-        address _tokenLock
-    );
+
     event WithdrawRequest(
-        uint256 indexed _pid,
         address indexed _beneficiary,
-        uint256 indexed _withdrawEnableTime
+        uint256 _withdrawEnableTime
     );
-    event Withdraw(address indexed user, uint256 indexed pid, uint256 shares);
-    event EmergencyWithdraw(address indexed user, uint256 indexed pid, uint256 amount);
+    event EmergencyWithdraw(address indexed user, uint256 assets, uint256 shares);
 
     modifier onlyFeeSetter() {
-        if (feeSetter != msg.sender) revert OnlyFeeSetter();
+        if (registry.feeSetter() != msg.sender) revert OnlyFeeSetter();
         _;
     }
 
-    modifier onlyCommittee(uint256 _pid) {
-        if (committees[_pid] != msg.sender) revert OnlyCommittee();
+    modifier onlyCommittee() {
+        if (committee != msg.sender) revert OnlyCommittee();
         _;
     }
 
     modifier onlyArbitrator() {
-        if (arbitrator != msg.sender) revert OnlyArbitrator();
+        if (registry.arbitrator() != msg.sender) revert OnlyArbitrator();
         _;
     }
 
     modifier noSafetyPeriod() {
-        //disable withdraw for safetyPeriod (e.g 1 hour) after each withdrawPeriod(e.g 11 hours)
+        HATVaultsRegistry.GeneralParameters memory generalParameters = registry.getGeneralParameters();
+        // disable withdraw for safetyPeriod (e.g 1 hour) after each withdrawPeriod(e.g 11 hours)
         // solhint-disable-next-line not-rely-on-time
         if (block.timestamp %
         (generalParameters.withdrawPeriod + generalParameters.safetyPeriod) >=
             generalParameters.withdrawPeriod) revert SafetyPeriod();
         _;
     }
 
-    modifier noActiveClaims(uint256 _pid) {
-        if (activeClaims[_pid] != 0) revert ActiveClaimExists();
+    modifier noActiveClaim() {
+        if (activeClaim.createdAt != 0) revert ActiveClaimExists();
+        _;
+    }
+
+    modifier activeClaimExists() {
+        if (activeClaim.createdAt == 0) revert NoActiveClaimExists();
         _;
     }
 
+    function initialize(
+        IRewardController _rewardController,
+        uint256 _vestingDuration,
+        uint256 _vestingPeriods,
+        uint256 _maxBounty,
+        BountySplit memory _bountySplit,
+        IERC20 _asset,
+        address _committee,
+        bool _isPaused
+    ) external initializer {
+        if (_vestingDuration > 120 days)
```

Comment:
perhaps use `setVestingParams` here do not duplicate these checks? (i.e. abstract it in an `_setVestingParams internal` and use that)

Reply:
Did that in the internal review PR

Reply:
https://github.com/hats-finance/hats-contracts/pull/281/commits/7b69095b2494db7e376089e5c38836050d5a97d0